src/CMakeLists.txt

```diff
@@ -54,6 +54,20 @@ IF (MSVC_IDE)
 	SET_SOURCE_FILES_PROPERTIES(win32/precompiled.c COMPILE_FLAGS "/Ycprecompiled.h")
 ENDIF ()
 
+# Collect sourcefiles
+FILE(GLOB LIBGIT2_SOURCES
```

Comment:
I want to point out that `file(GLOB …)` warns that :

>  If no CMakeLists.txt file changes when a source is added or removed then the generated build system cannot know when to ask CMake to regenerate.

This is somewhat unexpected, and sounds eerily like #4617, as the testsuite is also globbed. Maybe it would be worth to just flatten the list here ?

Reply:
(the commit message is missing a word at the end of the 3rd line)